src/zcbenchmarks.cpp

```diff
@@ -31,6 +31,31 @@
 #include "zcash/IncrementalMerkleTree.hpp"
 
 using namespace libzcash;
+// This method is based on Shutdown from init.cpp
+void shutdown_wallet()
+{
+    LogPrintf("%s: In progress...\n", __func__);
+    static CCriticalSection cs_Shutdown;
+    TRY_LOCK(cs_Shutdown, lockShutdown);
+    if (!lockShutdown)
+        return;
+
+if (pwalletMain)
+    pwalletMain->Flush(false);
+#ifdef ENABLE_MINING
+    GenerateBitcoins(false, NULL, 0);
+#endif
+    UnregisterNodeSignals(GetNodeSignals());
+    if (pwalletMain)
+       pwalletMain->Flush(true);
+
+UnregisterAllValidationInterfaces(); 
```

Comment:
Indentation

Reply:
We should also check whether there are any other validation interfaces that are registered besides the wallet. Ideally only the wallet would be unregistered, but if that's not possible then we need to make sure we re-register any others on restart.